packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_class_table.dart

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../../../../primitives/utils.dart';
+import '../../../../shared/common_widgets.dart';
+import '../../../../shared/table.dart';
+import '../../../../shared/table_data.dart';
+import '../../../../shared/utils.dart';
+import '../../memory_controller.dart';
+import 'allocation_profile_tracing_view_controller.dart';
+
+/// The default width for columns containing *mostly* numeric data (e.g.,
+/// instances, memory).
+const _defaultNumberFieldWidth = 80.0;
+
+class _TraceCheckBoxColumn extends ColumnData<TracedClass>
+    implements ColumnRenderer<TracedClass> {
+  _TraceCheckBoxColumn()
+      : super(
+          'Trace',
+          titleTooltip:
+              'Enable or disable allocation tracing for a specific type',
+          fixedWidthPx: scaleByFontFactor(55.0),
+          alignment: ColumnAlignment.left,
+        );
+
+  @override
+  bool get supportsSorting => false;
+
+  @override
+  Widget build(
+    BuildContext context,
+    TracedClass item, {
+    bool isRowSelected = false,
+    VoidCallback? onPressed,
+  }) {
+    final controller =
+        Provider.of<MemoryController>(context).allocationTracingController;
+    return Checkbox(
+      value: controller.isAllocationTracingEnabledForClass(item.cls),
+      onChanged: (value) async {
+        await controller.setAllocationTracingForClass(item.cls, value!);
+      },
+    );
+  }
+
+  @override
+  bool? getValue(TracedClass _) {
+    return null;
+  }
+
+  @override
+  int compare(TracedClass a, TracedClass b) {
+    return a.traceAllocations.boolCompare(b.traceAllocations);
+  }
+}
+
+class _AllocationTracingTableClassName extends ColumnData<TracedClass> {
+  _AllocationTracingTableClassName() : super.wide('Class');
+
+  @override
+  String? getValue(TracedClass stats) => stats.cls.name;
+
+  @override
+  bool get supportsSorting => true;
+}
+
+class _AllocationTracingTableInstances extends ColumnData<TracedClass> {
```

Comment:
nit on these column data class names. Since they are private, these column classes will only ever be used in this library, so can we get rid of the "_AllocationTracingTable" prefix in the name? e.g `_AllocationTracingTableClassName` --> `_ClassNameColumn`, `_AllocationTracingTableInstances` --> `_InstancesColumn`